### PR TITLE
web-ext: Include only production dependencies in build

### LINKED ISSUE #273083: web-ext missing git-rev-sync runtime dependency
### Describe the bug

It looks like [the upstream package has set a runtime dependency as a dev dependency](https://github.com/mozilla/web-ext/issues/2981), so the program crashes on every run.

### Steps To Reproduce

See [upstream bug report](https://github.com/mozilla/web-ext/issues/2981).

### Expected behavior

No crash.

### Notify maintainers

@dotlambda 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 23.11 (Tapir), 23.11.1209.933d7dc15509`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11, nixos-hardware"`
 - channels(username): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by l0b0**
## Description of changes

Closes #273083.

Big thanks to @Rob--W for [explaining](https://github.com/mozilla/web-ext/pull/2982)!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
Please make an upstream PR.

**Comment by l0b0** [comment]
> Please make an upstream PR.

I'm waiting to see if they acknowledge the [issue](https://github.com/mozilla/web-ext/issues/1349#issuecomment-1848369489).

**Comment by l0b0** [comment]
@dotlambda Turns out the problem was our build process, after all.

File: pkgs/development/tools/web-ext/default.nix [hunk 1]
```diff
@@ -16,6 +17,15 @@ buildNpmPackage rec {
 
   npmDepsHash = "sha256-KPBKUjCxva11w/E+Qhlx+1vikpCL7Hr9MiKenYHEVSU=";
 
+  preBuild = ''
+    export NODE_ENV=production
+  '';
```
Comment (dotlambda) [inline pkgs/development/tools/web-ext/default.nix:22]:
```suggestion
  env.NODE_ENV = "production";
```

File: pkgs/development/tools/web-ext/default.nix [hunk 2]
```diff
@@ -1,8 +1,9 @@
 { lib
 , buildNpmPackage
 , fetchFromGitHub
+, runCommand
+, web-ext
 }:
-
```
Comment (dotlambda) [inline pkgs/development/tools/web-ext/default.nix:5]: Please don't remove this line.

Comment (l0b0) [reply pkgs/development/tools/web-ext/default.nix:5]: Done.

File: pkgs/development/tools/web-ext/default.nix (hunk 1 above)
Comment (l0b0) [reply pkgs/development/tools/web-ext/default.nix:23]: No, this breaks the build.

Comment (dotlambda) [reply pkgs/development/tools/web-ext/default.nix:23]: Does `npmBuildFlags = [ "--production" ]` work?

Comment (l0b0) [reply pkgs/development/tools/web-ext/default.nix]: Yep, thanks! I should probably add you as co-author by now, do you want me to?

**Comment by dotlambda** [comment]
@l0b0 I forgot to ask if you want to add yourself as maintainer.

**Comment by l0b0** [comment]
> @l0b0 I forgot to ask if you want to add yourself as maintainer.

I'm not entirely sure yet. I have a minor hobby [project](https://gitlab.com/engmark/insecure-links-highlighter/) where I'd like to use this, but I've no experience with upstream other than this issue.

**Pull request merged**